examples/custom-node/src/engine.rs

```diff
@@ -0,0 +1,384 @@
+    fn parent_beacon_block_root(&self) -> Option<revm_primitives::B256> {
+        self.inner.parent_beacon_block_root()
+    }
+
+    fn timestamp(&self) -> u64 {
+        self.inner.timestamp()
+    }
+
+    fn withdrawals(&self) -> Option<&Vec<alloy_eips::eip4895::Withdrawal>> {
+        self.inner.withdrawals()
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct CustomPayloadBuilderAttributes {
+    inner: OpPayloadBuilderAttributes<OpTransactionSigned>,
+    _extension: u64,
+}
+
+impl PayloadBuilderAttributes for CustomPayloadBuilderAttributes {
+    type RpcPayloadAttributes = CustomPayloadAttributes;
+    type Error = alloy_rlp::Error;
+
+    fn try_new(
+        parent: revm_primitives::B256,
+        rpc_payload_attributes: Self::RpcPayloadAttributes,
+        version: u8,
+    ) -> Result<Self, Self::Error>
+    where
+        Self: Sized,
+    {
+        let CustomPayloadAttributes { inner, extension } = rpc_payload_attributes;
+
+        Ok(Self {
+            inner: OpPayloadBuilderAttributes::try_new(parent, inner, version)?,
+            _extension: extension,
+        })
+    }
+
+    fn parent(&self) -> revm_primitives::B256 {
+        self.inner.parent()
+    }
+
+    fn parent_beacon_block_root(&self) -> Option<revm_primitives::B256> {
+        self.inner.parent_beacon_block_root()
+    }
+
+    fn payload_id(&self) -> alloy_rpc_types_engine::PayloadId {
+        self.inner.payload_id()
+    }
+
+    fn prev_randao(&self) -> revm_primitives::B256 {
+        self.inner.prev_randao()
+    }
+
+    fn suggested_fee_recipient(&self) -> revm_primitives::Address {
+        self.inner.suggested_fee_recipient()
+    }
+
+    fn timestamp(&self) -> u64 {
+        self.inner.timestamp()
+    }
+
+    fn withdrawals(&self) -> &alloy_eips::eip4895::Withdrawals {
+        self.inner.withdrawals()
+    }
+}
+
+impl BuiltPayload for CustomBuiltPayload {
+    type Primitives = CustomNodePrimitives;
+
+    fn block(&self) -> &SealedBlock<<Self::Primitives as NodePrimitives>::Block> {
+        self.0.block()
+    }
+
+    fn executed_block(&self) -> Option<ExecutedBlockWithTrieUpdates<Self::Primitives>> {
+        self.0.executed_block()
+    }
+
+    fn fees(&self) -> U256 {
+        self.0.fees()
+    }
+
+    fn requests(&self) -> Option<alloy_eips::eip7685::Requests> {
+        self.0.requests()
+    }
+}
+
+impl From<CustomBuiltPayload>
+    for alloy_consensus::Block<<CustomNodePrimitives as NodePrimitives>::SignedTx>
+{
+    fn from(value: CustomBuiltPayload) -> Self {
+        value.0.into_sealed_block().into_block().map_header(|header| header.inner)
+    }
+}
+
+impl From<CustomBuiltPayload> for ExecutionPayloadV1 {
+    fn from(value: CustomBuiltPayload) -> Self {
+        Self::from_block_unchecked(value.block().hash(), &value.into())
+    }
+}
+
+impl From<CustomBuiltPayload> for ExecutionPayloadV2 {
+    fn from(value: CustomBuiltPayload) -> Self {
+        Self::from_block_unchecked(value.block().hash(), &value.into())
+    }
+}
```

Comment:
with recent engine rpc modifcations we're likely more flexible when it comes to defining the enginetypes so I think we can exclude these conversion from this pr for now.

and instead just implement our own smol engineAPI equivalent API that maybe just has one single newPayload getPayload impl